### DISCUSSION TITLE
How to set a title per note so the Obsidian Graph looks nice?

### DISCUSSION THREAD

**Discussion by UberChili** [epwalsh/obsidian.nvim#645] [Q&A] [answered]
I like how the default configuration automatically adds an ID to new notes, but if I go to the Obsidian App and try to view my graph, it looks quite ugly as it only shows the ID's of the notes. Is there a way in the configuration, or in Obsidian, to make it show the aliases or for the notes YAMLs to include a title? I am imagining something like Org Mode's "#+Title:"

**Comment by r00t3dd** [comment, marked as answer]
Hey @UberChili, you can achieve this by adding note_path_func callback in your obsidian.nvim settings. For example:

```lua
      -- Optional, customize how note file names are generated given the ID, target directory, and title.
      ---@param spec { id: string, dir: obsidian.Path, title: string|? }
      ---@return string|obsidian.Path The full path to the new note.
      note_path_func = function(spec)
        -- This is equivalent to the default behavior.
        local path = spec.dir / tostring(spec.id) .. "+" .. tostring(spec.title)
        return path:with_suffix(".md")
      end,
```

https://github.com/epwalsh/obsidian.nvim/blob/e170641143704068233ba1d301e75684b67fc07c/README.md?plain=1#L320

**Reply by UberChili** [reply]
Thank you, this helped me a little, even though I don't completely understand still.
So when I put this inside opts={}; nvim shows me "undefined type or alias 'obsidian.Path' ", and also an "undefined field 'with_suffix'. Other than that, I modified the lines to:
```lua
local path = spec.title .. "+" .. spec.id
return path
```

I feel like I am doing something wrong.
Thanks for the patience!
